pkg/acquisition/modules/http/http.go

```diff
@@ -0,0 +1,446 @@
+		}
+		if hc.BasicAuth.Username == "" {
+			return errors.New("username is required")
+		}
+		if hc.BasicAuth.Password == "" {
+			return errors.New("password is required")
+		}
+	case "headers":
+		if hc.Headers == nil {
+			return errors.New("headers is required")
+		}
+	case "mtls":
+		if hc.TLS == nil || hc.TLS != nil && hc.TLS.CaCert == "" {
+			return errors.New("ca_cert is required")
+		}
+	default:
+		return errors.New("invalid auth_type: must be one of basic_auth, headers, mtls")
+	}
+
+	if hc.TLS != nil {
+		if hc.TLS.ServerCert == "" {
+			return errors.New("server_cert is required")
+		}
+		if hc.TLS.ServerKey == "" {
+			return errors.New("server_key is required")
+		}
+	}
+
+	if hc.MaxBodySize != nil && *hc.MaxBodySize <= 0 {
+		return errors.New("max_body_size must be positive")
+	}
+
+	/*
+		if hc.ChunkSize != nil && *hc.ChunkSize <= 0 {
+			return errors.New("chunk_size must be positive")
+		}
+	*/
+
+	if hc.CustomStatusCode != nil {
+		statusText := http.StatusText(*hc.CustomStatusCode)
+		if statusText == "" {
+			return errors.New("invalid HTTP status code")
+		}
+	}
+
+	return nil
+}
+
+func (h *HTTPSource) Configure(yamlConfig []byte, logger *log.Entry, MetricsLevel int) error {
+	h.logger = logger
+	h.metricsLevel = MetricsLevel
+	err := h.UnmarshalConfig(yamlConfig)
+	if err != nil {
+		return err
+	}
+
+	if err := h.Config.Validate(); err != nil {
+		return fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	return nil
+}
+
+func (h *HTTPSource) ConfigureByDSN(string, map[string]string, *log.Entry, string) error {
+	return fmt.Errorf("%s datasource does not support command-line acquisition", dataSourceName)
+}
+
+func (h *HTTPSource) GetMode() string {
+	return h.Config.Mode
+}
+
+func (h *HTTPSource) GetName() string {
+	return dataSourceName
+}
+
+func (h *HTTPSource) OneShotAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	return fmt.Errorf("%s datasource does not support one-shot acquisition", dataSourceName)
+}
+
+func (h *HTTPSource) CanRun() error {
+	return nil
+}
+
+func (h *HTTPSource) GetMetrics() []prometheus.Collector {
+	return []prometheus.Collector{linesRead}
+}
+
+func (h *HTTPSource) GetAggregMetrics() []prometheus.Collector {
+	return []prometheus.Collector{linesRead}
+}
+
+func (h *HTTPSource) Dump() interface{} {
+	return h
+}
+
+func (hc *HttpConfiguration) NewTLSConfig() (*tls.Config, error) {
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: hc.TLS.InsecureSkipVerify,
+	}
+
+	if hc.TLS.ServerCert != "" && hc.TLS.ServerKey != "" {
+		cert, err := tls.LoadX509KeyPair(hc.TLS.ServerCert, hc.TLS.ServerKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load server cert/key: %w", err)
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+
+	if hc.AuthType == "mtls" && hc.TLS.CaCert != "" {
+		caCert, err := os.ReadFile(hc.TLS.CaCert)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read ca cert: %w", err)
+		}
+		caCertPool := x509.NewCertPool()
```

Comment:
Some users may want to load from system pool?

ref: https://github.com/crowdsecurity/go-cs-bouncer/blob/9e16e182103d8f9ac30352bff0c1eca27b0619ba/tls.go#L11-L35